File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.ends_with?(input, "?") -> "Sure."
      String.match?(input, ~r/^\s*$/) -> "Fine. Be that way!"
      String.upcase(input) == input &&
        String.match?(input, ~r/\p{Lu}/) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end

File: javascript/bob/bob.js
//
// This is only a SKELETON file for the "Bob" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Bob = function() {};

Bob.prototype.hey = function(input) {
  switch(true) {
    case input.toUpperCase() == input && /[A-Z]/.test(input):
      return 'Whoa, chill out!'
    case /\?$/.test(input):
      return 'Sure.';
    case /^\s*$/.test(input):
      return 'Fine. Be that way!';
    default:
      return 'Whatever.'
  }
};

module.exports = Bob;
